packages/components/cascader/__tests__/cascader.test.tsx

```diff
@@ -445,4 +445,24 @@ describe('Cascader.vue', () => {
     await nextTick()
     expect(dropdown?.style.display).not.toBe('none')
   })
+
+  test('height should be changed by size when multiple', async () => {
+    const cascaderSize = ref('small')
+    const props = { multiple: true }
+    const wrapper = _mount(() => (
+      <Cascader props={props} size={cascaderSize.value} />
+    ))
+    await nextTick()
+    const inputEl = wrapper.find('input').element as HTMLElement
+    const sizeMap: Record<string, number> = {
+      small: 24,
+      default: 32,
+      large: 40,
+    }
+    for (const size in sizeMap) {
+      cascaderSize.value = size
+      await nextTick()
+      expect(inputEl.style.height).toEqual(`${sizeMap[size] - 2}px`)
```

Comment:
@chenxch  this way~ I am used the `height` of style to judge if it is changed when size changed, but the height is always `null` because it didn't be calculated right.

Reply:
😂 我一直没用 mock 主要感觉用了 mock 这个单测就没啥作用了，mock完 input.offsetHeight 的值就是自己设置判断跟自己相等了

Reply:
mock这种只能说属于是一个假hack方式了（因为下一个修改的发现这边测试过去就跑来改这边的值）

Reply:
之前select其实也存在这个问题，只不过是修复，感觉可以像select看齐调整。 large那个看样子是盖到下拉箭头了吧，黄色色块是说上下间距问题么？

Reply:
是，就是盖到下拉箭头了。改一下 `large` 时候的 `right` 值应该就好了（只是我不知道要改成多少）：
![image](https://user-images.githubusercontent.com/30046649/229570304-ee4752c5-0441-4ead-957d-889fb985fd95.png)

看了下 `select` 多选时候的，应该是一个计算了 `max-width`， 跟这个有点不一样

Reply:
按照箭头左右两边空间相等来改的话 `large` 的时候好像又太宽了：
![image](https://user-images.githubusercontent.com/30046649/229573921-7378c345-21da-44d7-a551-cac60de49e12.png)

Reply:
单元测试这边是因为jsdom这种不会生成实际的元素，导致offsetHeight这种获取不到，我们可以通过beforAll，或者是spyOn的方式去处理这个

Reply:
beforeAll这个无法动态修改，spyOn修改了作用不到jsdom。。。

Reply:
> 单元测试这边是因为jsdom这种不会生成实际的元素，导致offsetHeight这种获取不到，我们可以通过beforAll，或者是spyOn的方式去处理这个

🙈你也太肝了吧。感觉是jsdom没有做到渲染那层，所以拿css相关的属性都不太靠谱，我今天也试试这两个看看有没搞头

Reply:
我直接模拟真实环境下的 `height` 你看看行不：
![image](https://user-images.githubusercontent.com/30046649/229740916-9ddd869f-8028-4e09-974e-414196cff34d.png)